app/src/main/java/org/mozilla/focus/search/MultiselectSearchEngineListPreference.java

```diff
@@ -48,4 +52,36 @@ protected void updateDefaultItem(CompoundButton defaultButton) {
         }
         return engineIdSet;
     }
+
+    // Whenever an engine is checked or unchecked, we notify the menu
+    public void bindEngineCheckboxesToMenu() {
+        for (int i = 0; i < searchEngineGroup.getChildCount(); i++) {
+            final CompoundButton engineButton = (CompoundButton) searchEngineGroup.getChildAt(i);
+            engineButton.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+                @Override
+                public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+                    ((Activity) getContext()).invalidateOptionsMenu();
+                }
+            });
+        }
+        setSearchEnginesBindingToMenu(true);
+    }
+
+    public boolean atLeastOneEngineChecked() {
+        for (int i = 0; i < searchEngineGroup.getChildCount(); i++) {
+            final CompoundButton engineButton = (CompoundButton) searchEngineGroup.getChildAt(i);
+            if (engineButton.isChecked()) {
+               return true;
+            }
+        }
+        return false;
+    }
+
+    private void setSearchEnginesBindingToMenu(boolean binding) {
+        this.searchEnginesBindingToMenu = binding;
+    }
+
+    public boolean isSearchEnginesBindingToMenu() {
```

Comment:
Is this so that we don't repeatedly bind an onCheckedChangeListener? Since this doesn't actually depend on the MenuItem anymore (it just sends an "invalidate" message) we can just move this into the construction of the Preference.